src/w3iProxy/w3iChatFacade.ts

```diff
@@ -35,6 +47,58 @@ class W3iChatFacade implements W3iChat {
 
     const ProviderClass = this.providerMap[this.providerName]
     this.provider = new ProviderClass(this.emitter)
+
+    // Discuss expiry of messages
+    this.messageReplaySubject = new ReplaySubject(undefined, ONE_DAY / 2)
+    this.handleMessagePublishing()
+  }
+
+  private handleMessagePublishing() {
+    // Stop trying to resend the message after 3 tries.
+    this.messageReplaySubject.pipe(filter(params => params.count < 3)).subscribe(params => {
+      from(this.provider.message(params))
+        .pipe(
+          timeout({
+            first: this.messageSendTimeout,
+            with: () => throwError(() => new Error())
```

Comment:
Should this throw with a concrete error?

Reply:
Right now, we do nothing with the error so I don't see the need to, we could possibly change this if in the future `provider.request` in core threw errors in relation to connection, but for now, there's nothing to do with the errors other than catch and retry